添加了Sydney搭建微信聊天机器人，基于微软Bing

JayGarland opened the pull request.

JayGarland committed:
Please follow instructions in Readme.md!

JayGarland committed:
update cookies.json on list

JayGarland committed:
add feedback of exception info, prompt setting

JayGarland committed:
add function of clearall session conversation

JayGarland committed:
change query in conversation

JayGarland committed:
change respective model for sydney

JayGarland committed:
Update .gitignore

resphinas commented:
建议优化一下可读性太差了
1.sydney_process 里面有个写死的你自己的代理地址
2.                                    async for secresponse in para:
                                        # print(secresponse)
                                        if secresponse["type"] == 1 and "messages" in secresponse["arguments"][0]:
                                            message = secresponse["arguments"][0]["messages"][0]
                                            msg_type = message.get("messageType")
                                            if msg_type is None:
                                                if message.get("contentOrigin") == "Apology":
                                                    failed = True
                                                    break
                                                else:
                                                    replied = True
                                                    reply = ""                   
                                                    reply += remove
好几年没见到这么多层if的代码了
3.有个bleach 依赖要装
4.建议整理下以来文件，你指定的加速地址很差，可以改用清华源
5.cookie文件放到目录的cookies.json中与原项目架构不符，建议加入config.json的生态
6.sydney_process 头部有n多个没用上的包，冗余代码。

JayGarland committed:
add debug option, fix daily auth issue

JayGarland committed:
update prompt, improve readability, keep it replying until the end

JayGarland committed:
Update session_manager.py

JayGarland committed:
handle exception,

JayGarland committed:
progressive update

JayGarland committed:
Update config.py

JayGarland committed:
Update README.md

JayGarland committed:
Update config.py

JayGarland commented:
> 一些建议 1.sydney_process 里面有个写死的代理地址 2. if 嵌套严重 可读性差 3.依赖文件指定的加速pypi源效果很差，可以改用清华源 4.cookie文件放到目录的cookies.json中与原项目架构不符，建议加入config.json 5.多个没用上的冗余包，冗余代码。 6.循环打印，狂刷控制台，可改成非流式输出&清理一下print语句 7.个性化语句太明显，建议改成通用点的prompt

谢谢

JayGarland committed:
update prompt, bot_statement

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
update efficient prompt, add roleplay

JayGarland committed:
Update Sydney_proess.py

JayGarland committed:
update wework

JayGarland committed:
add Type of messages

JayGarland committed:
at_list msg

JayGarland committed:
global list

JayGarland committed:
prompt

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
update time limit, and retry

JayGarland committed:
set None if no proxy

JayGarland committed:
Update Sydney_proess.py

JayGarland committed:
Update chat_channel.py

JayGarland committed:
prompt

JayGarland committed:
add two preset roles

JayGarland committed:
retry fun

JayGarland committed:
add suffix at the end of sydney_reply, which converts it to dict

JayGarland committed:
improve reply format in exception

JayGarland committed:
timer limit, prompt

JayGarland committed:
try exception

JayGarland committed:
Update dungeon.py

JayGarland committed:
update Zh

JayGarland committed:
add trigger en

JayGarland committed:
add summary prompt

JayGarland committed:
throttled handle

JayGarland committed:
Update Sydney_proess.py

JayGarland committed:
Merge branch 'master' into Sydney

JayGarland committed:
update reply algorithm

JayGarland committed:
change context label to Sydney

JayGarland committed:
change wss

JayGarland committed:
Merge branch 'master' into Sydney

JayGarland committed:
Merge branch 'master' into Sydney

JayGarland committed:
sharing tag message (pass)

JayGarland committed:
Merge branch 'master' into Sydney

JayGarland committed:
Merge branch 'master' into Sydney

JayGarland committed:
Merge branch 'master' into Sydney

JayGarland committed:
Update Sydneyreply.py

raJayoux committed:
Merge branch 'master' into Sydney

JayGarland committed:
add retry

JayGarland committed:
Merge branch 'master' into Sydney

raJayoux committed:
Merge branch 'master' into Sydney

JayGarland committed:
Update const.py

JayGarland committed:
update header info, init image create

JayGarland committed:
One issue report, now the img recognize process is unclear, will lead to bot crush, not my fault, maybe because the "self" variable

JayGarland committed:
update timeout

JayGarland committed:
sydney -> assistant

JayGarland committed:
discard sydneyprocess to sydneyreply

JayGarland committed:
merge and custom update from linkaiBot

JayGarland committed:
add debugs

JayGarland committed:
add debuggs

JayGarland committed:
add image recognize, but currently when the img is too big then the server give a refuse

JayGarland committed:
established the image realization

JayGarland committed:
update webPagefetch

JayGarland committed:
update PPT, Docx, Word file Read

JayGarland committed:
extend expiredTimeout

raJayoux committed:
adjust position of link and file in the precontext

raJayoux committed:
improve webpageinfo

JayGarland committed:
add exception for the file type

JayGarland committed:
fixed generating unfocused response in the next

JayGarland committed:
wrapped error exception when throttled, 443, captcha

JayGarland committed:
Update godcmd.py

JayGarland committed:
add role variety

JayGarland committed:
fix a minor mistake in exception handling,

When you write something like this:

elif "throttled" or "Throttled" in str(e):
You are actually checking two separate conditions: "throttled" and "Throttled" in str(e). The first one is always True, because any non-empty string is considered True in Python. The second one is only True if the string e contains the word "Throttled". So, the whole expression is always True, regardless of what e is. That's why your code always returns the handle of detecting throttled, even when e is None. 😬

JayGarland committed:
fix minor error in str(e)

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
add reset sydney in godmod

JayGarland committed:
fix redundant exceptions, fix botstatement error when no reply

JayGarland committed:
customized sydney Readme!

JayGarland committed:
todo stop running async process

JayGarland committed:
when no reply then ask again

JayGarland committed:
when wrap with bot statement, try again if no reply

JayGarland committed:
wrapped out send out exception error to users

JayGarland committed:
additional instructions added for plugin

JayGarland committed:
adjust position of imgurl, and make it none with condition, so the imgurl won't be none in the next try

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
add emoji in output, and switch output language based on user input

JayGarland committed:
replaced the alphabet_only function with an except_chinese_chat

JayGarland committed:
big update, fixed attempting secResponse, and improve performance, Sydney now can output the same language as the user

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
fix unbounded local variable

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
handle async response to let user stop at any time

JayGarland committed:
Update README.md

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
Update README.md

JayGarland committed:
reset currently responding task, and prepare for send out message when the bot sees user's message

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
add emoji in tip message, add tip message to tell if the reply is in process

JayGarland committed:
Update README.md

JayGarland committed:
Update README.md

JayGarland committed:
Update README.md

JayGarland committed:
todo image create

JayGarland committed:
todo

JayGarland committed:
todo imageCreate

JayGarland committed:
feat: update create conversation params

JayGarland committed:
add a lazy stop trigger to stop acting, but user needs to call the trigger again to stop the current instance

JayGarland committed:
user can stop even in the plugin game

JayGarland committed:
the same action as the previous commit but for the role plugin

JayGarland committed:
adjust webinfo and fileinfo in the precontext

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
ignore particular instructions in precontext

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
feat: general optimization for creating conversation params

JayGarland committed:
fixed webpage,file info in a conversation

JayGarland committed:
feat: fixed file,web info in a conversation

JayGarland committed:
fix webpageinfo, fileinfo in a single conversation

JayGarland committed:
improve perfermance of output but now it might have  hallucination

JayGarland committed:
add searchQuery, off internalsearch, add gptid para in ask stream

JayGarland committed:
revert links in output

JayGarland committed:
Update sydney.py

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
add planning updates and update fixes

JayGarland committed:
Update README.md

JayGarland committed:
remove redundant additional instructions

JayGarland committed:
adjust position between the restMessage  and web,file info in precontext

JayGarland committed:
creaitveclassic, nosearch, image search

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
use creative

JayGarland committed:
Merge remote-tracking branch 'upstream/master' into Sydney

JayGarland committed:
continous the reply context when been interrupted

JayGarland committed:
revert and update timeout handle in askstream, create conversation

JayGarland committed:
grey test, timeout error fix, revert wssurl connect

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
comment our image create func

JayGarland committed:
remove bot statement in precontext

JayGarland committed:
complete cut botstatement in session_message

JayGarland committed:
remove repeating words issue in output

JayGarland committed:
show tip when over max tries, retry ask_stream if got a bad output

 retry when there are 12 times the same pair of chinese characters in output

JayGarland committed:
todo retry tip

JayGarland committed:
comment out incoming message info

JayGarland committed:
feat updates!

using query instead of custom ask_string, improve the tip message when retry times maxed, such as clip the question words in 10 words, adjust prompt, add Bing filter and try again if jailbreak failed, adjust botstatement occurance only at the first message then it will not be added later, adjusts the detected times of repeating words in the output

JayGarland committed:
switch option set

JayGarland committed:
add revoke edit instruction when chat with the bot, improve precontext content by pop the message if there is user instructions

JayGarland committed:
botpreset text ignore

JayGarland committed:
Update session_manager.py

JayGarland committed:
clip disclaimer in message

JayGarland committed:
using fluxsydney

JayGarland committed:
add todos

JayGarland committed:
change optionset, add todos, change locale, add tip message if timedout in image upload process

JayGarland committed:
remove triggered message in session.messages

JayGarland committed:
update removing triggered message from _chat, fix retry chat param, check if exceed amount of repeating words while in generation of reply

JayGarland committed:
adjust pass tip message of memory temp clean and cancel func

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
rearrange the code and change reply variable name to passivereply in func reply, update the detection of tip message in reply func , add todos

JayGarland committed:
add a tip message when bot received the user message

JayGarland committed:
adjust bot statement

JayGarland committed:
del old sydney_process

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
if upload image failed, add suffix of imgsentfailed tip message at the end of reply

JayGarland committed:
update tipmessage text when bot received user msg

JayGarland committed:
add suggestions at the end of replycontent and adjust botstatement

JayGarland committed:
Update sydney.py

JayGarland committed:
add error info in imgfailedmsg

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
Update sydney.py

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
remove mini funtion type in contextfiletype

JayGarland committed:
add a pull request from the master which reply in single chat only if keyword

JayGarland committed:
add passivereply for patpat in context

JayGarland committed:
add hello message when added a new friend, adjust the original patpat

JayGarland committed:
add keyword checks for English version wechat , but still have todos

JayGarland committed:
bot won't show tipmsg when certain keywords in content, add todo make the certain instruction loaded in the config.json instead writing it in the code

JayGarland committed:
no need to calculate tokens for sydneybot

JayGarland committed:
Update chat_channel.py

JayGarland committed:
add todo, add more keywords and english tipmessage

JayGarland committed:
add config for sydneybot， add todo

JayGarland committed:
load sydneykeywords from config file, add intervals for image receive

JayGarland committed:
load bot statemenet, bot prompt setting from configfile, sendout retry info when retry

JayGarland committed:
adjust keywords, sydneycredit

JayGarland committed:
when repeatingg question then pop the users msg, add credit, statement

JayGarland committed:
fix when no msg to revoke, pop msg when in exception, raise expception when certain words in reply, bot replies retrying info when retrying

JayGarland committed:
done todo

JayGarland committed:
revert ignore tokens

JayGarland committed:
remove commented

JayGarland committed:
revert sydney in dungeon

JayGarland committed:
add promote images in the first message

JayGarland committed:
#todo added bug, when contain added string then it will go accept_friend type if tickle the bot

JayGarland committed:
fix added bug, when contain added string then it will go accept_friend type if tickle the bot

JayGarland committed:
debug when bot reiceived the same message from the other user it now won't reply it's a repeating word when the lastquery from other user equals to the current query, located the first message by session.messages

JayGarland committed:
fix conn aborted problem when sending promoteinfo

JayGarland committed:
add qrcode image like donate and id

JayGarland committed:
done promote

JayGarland committed:
add new keyword detect in bing filter

JayGarland committed:
fix type error

JayGarland committed:
cancel pop message when except, adjust param of compressing image size

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
adjust imgfailedmsg content and adjust the tip message position in reply

JayGarland committed:
load debug option from sydney file to config file

JayGarland committed:
add info of the repeating question, assign the lastquery to the second last user message when except happened in _chat, add debug info when there is an exception in reply func, also add debug info in _chat

JayGarland committed:
adjust dev creidt  position and put it in a new message and send out

JayGarland committed:
adjust retry when Conn aborted

JayGarland committed:
adjust retry logic, set ignore disclaimer to optional, cuz sometimes it will influence the whole context format

JayGarland committed:
update promote info, set qrpayimg optional

JayGarland committed:
add customized persona for different chat group or people, the bot now can be whoever you wantmore flexiable!

JayGarland committed:
update location and optionset

JayGarland committed:
revert

JayGarland committed:
update optionset

JayGarland committed:
Revert "update optionset"

This reverts commit 6f33b78380761a92b3dc58a478a37c8b71c36366.

JayGarland committed:
recover messageid

JayGarland committed:
add nosearch, custom botstatement, suggest params in config file

JayGarland committed:
fix suggest tip

JayGarland committed:
update Sydney core

JayGarland committed:
Revert "update Sydney core"

This reverts commit 8e5d1a018c6745dca97b33d197e544b64aa38954.

JayGarland committed:
feat: bot process core update, add utils and Chathub from reedgegpt

JayGarland committed:
Update upload_image.py

JayGarland committed:
import chatbot from reedgegpt,

moved v1utils to another folder, discarded old version img upload process, wrap v1sydney ask_stream, now using ask_stream from the reedge chatbot

JayGarland committed:
change locale, add repeating words filter

JayGarland committed:
add no_link option

JayGarland committed:
add no_link resp option, and change the response_text getting value from the text, directly not inline.text

JayGarland committed:
add todos, add tip message when exception happens

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
no search

JayGarland committed:
adjust tip msg process when in exception

JayGarland committed:
update exception info

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
add continous chat in single convid todos

JayGarland committed:
chore and add file process todo

JayGarland committed:
add bot statement v2

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
update exception reply process

JayGarland committed:
rearrange util code

JayGarland committed:
append tip msg to the reply when bot got apology

JayGarland committed:
add voice reply

JayGarland committed:
Update Sydneyreply.py

JayGarland committed:
Merge branch 'Sydney' of https://github.com/JayGarland/chatgpt-on-wechat into Sydney

JayGarland committed:
Update toggle voice output

JayGarland committed:
imageupload update

JayGarland committed:
switch voicespecies in different persona, replace bot.close with aiosession.close, debug reply none to str error

JayGarland committed:
add FIXME if try creating a channel wechaty instead of itchat

there are much things to do... to achieve the voice sending msg instead of sending mp3 files..

JayGarland committed:
add more info

JayGarland committed:
relocate the bot close

JayGarland committed:
Update chathub.py
